apache/superset
# feat(ui-core): implement drilldown

Add hierarchical traversability to pie charts and bar charts.

### SUMMARY
This is an attempt at bringing https://github.com/apache/superset/pull/14688 and https://github.com/apache-superset/superset-ui/pull/1120 up-to-date with the latest Superset mono repo and extending the feature to work both with echarts/pie and nvd3/distbar.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
See video on https://github.com/apache/superset/pull/14688

### TESTING INSTRUCTIONS
Create either a bar chart or pie chart.  In the chart editor, select "ENABLE DRILL DOWN."  Add multiple columns to the "GROUP BY" field that have a natural hierarchy to them and ensure the order of columns listed reflects the intended hierarchy.  Add an appropriate metric (usually COUNT(id) or similar).  Observe that only the first column in the hierarchy is reflected in the rendered chart.  Observe that clicking one field in the chart adds the selected field to the applied filters and that the second column in the "GROUP BY" hierarchy is now displayed.  Observe that selecting a field when the chart is at its deepest level in the hierarchy clears all selected filters and returns the chart to the original state.  Observe that saving the chart and adding it to a dashboard results in identical functionality as described above.

### ADDITIONAL INFORMATION


- [x] Required feature flags: DASHBOARD_DRILL_DOWN
